File: springdoc-openapi-tests/springdoc-openapi-javadoc-tests/src/test/java/test/org/springdoc/api/app174/ExampleController.java
package test.org.springdoc.api.app174;

import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * The Example Controller
 */
@RestController
class ExampleController {

	@PostMapping
	public Test post(){
		return null;
	}

}

File: springdoc-openapi-tests/springdoc-openapi-kotlin-mvc-tests/src/test/kotlin/test/org/springdoc/api/app8/ExampleController.kt
package test.org.springdoc.api.app8

import io.swagger.v3.oas.annotations.Parameter
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

data class Greeting(val greeting: String)

@RestController
interface ExampleController {
    @GetMapping("/")
    fun greet(@RequestParam name: String?): Greeting

	@GetMapping("/test")
	fun test(@RequestParam @Parameter(required = true) name: String?): Greeting
}
